### PR TITLE
[spinel] add CSL-related properties

### PR DESCRIPTION
**Pull request by ankuns**
This PR adds spinel properties for NCP built with `OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE`
Such NCP will have also a new capability SPINEL_CAP_THREAD_CSL_RECEIVER

### CODE REVIEW

**Review by brianljt — APPROVED** [review]
Thank you for the PR. LGTM 👍

File: src/lib/spinel/spinel.h [hunk 1]
```diff
@@ -2897,6 +2898,32 @@ enum
      */
     SPINEL_PROP_THREAD_NEW_DATASET = SPINEL_PROP_THREAD_EXT__BEGIN + 40,
 
+    /// MAC CSL Period
+    /** Format: `S` - MTD build only
```
Comment (Irving-cl) [inline src/lib/spinel/spinel.h:2902]: These properties are for CSL Receiver. I think FTD can also work as a CSL Receiver.

**Review by Irving-cl — APPROVED** [review]
LGTM :+1:

File: src/lib/spinel/spinel.h (hunk 1 above)
Comment (ankuns) [reply src/lib/spinel/spinel.h:2902]: As stated in the spec (Thread 1.2.0 spec June 18, 2019).
Chapter 11.1.1: _An FTD may act as an Active Router, a REED, or a FED. An MTD may act as a MED, SED,
SSED or BED._
CSL Receiver is a feature for SSED, so MTD only.
Do you think that it makes sense for FTD which is never rx-off-when-idle ?

Comment (Irving-cl) [reply src/lib/spinel/spinel.h:2902]: FTD, MTD refers to device capability. Generally, MTD is has a subset function of FTD. Usually we don't use FTD to act as SED or SSED. But it doesn't mean it cannot act as a SED. For example, we have two FTDs. We can also make one of them act as SED and enjoy low power consumption.

**Comment by ankuns** [comment]
@googlebot I signed it!

**Review by edmont — APPROVED** [review]

**Comment by ankuns** [comment]
@jwhui There is some CI issue. I have run `git commit --amend` and  `git push` several times to trigger CI. Checks are failing randomly however I have not modified the code.

**Review by abtink — APPROVED** [review]
LGTM. 👍

**Comment by jwhui** [comment]
> @jwhui There is some CI issue. I have run `git commit --amend` and `git push` several times to trigger CI. Checks are failing randomly however I have not modified the code.

GitHub Action is having some issues. It seems the relevant checks have passed now.

**Review by jwhui — APPROVED** [review]
Thanks! 👍🏼

**Pull request merged**